seed/static/seed/js/controllers/inventory_map_controller.js

```diff
@@ -0,0 +1,409 @@
+              spinner_utility) {
+      spinner_utility.show();
+
+      $scope.inventory_type = $stateParams.inventory_type;
+      $scope.data = inventory.results;
+
+      $scope.geocoded_data = _.filter($scope.data, function(building) {
+        return building.long_lat
+      });
+      $scope.ungeocoded_data = _.filter($scope.data, function(building) {
+        return !building.long_lat
+      });
+
+      // Map
+      var base_layer = new ol.layer.Tile({
+        source: new ol.source.Stamen({
+          layer: 'terrain'
+        }),
+        zIndex: 0,  // Note: This is used for layer toggling.
+      });
+
+      // Define buildings source - the basis of layers
+      var buildingPoint = function (building) {
+        var format = new ol.format.WKT();
+
+        var long_lat = building.long_lat
+        var feature = format.readFeature(long_lat, {
+          dataProjection: 'EPSG:4326',
+          featureProjection: 'EPSG:3857'
+        });
+
+        feature.setProperties(building)
+        return feature
+      };
+
+      var buildingSources = function (records = $scope.geocoded_data) {
+        var features = _.map(records, buildingPoint);
+
+        return new ol.source.Vector({ features: features });
+      };
+
+      // Points/clusters layer
+      var clusterPointStyle = function (size) {
+        // TODO revisit this with a larger dataset
+        // point radius of each cluster icon is relative to size and has a min and max.
+        var relative_radius = 10 + Math.min(7, size/50)
+        return new ol.style.Style({
+          image: new ol.style.Circle({
+            radius: relative_radius,
+            stroke: new ol.style.Stroke({
+              color: '#fff'
+            }),
+            fill: new ol.style.Fill({
+              color: '#3399CC'
+            })
+          }),
+          text: new ol.style.Text({
+            text: size.toString(),
+            fill: new ol.style.Fill({ color: '#fff' })
+          })
+        });
+      };
+
+      var singlePointStyle = function () {
+        return new ol.style.Style({
+          image: new ol.style.Icon({
+            src: urls.static_url + "seed/images/map_pin.png",
+            scale: 0.05,
+            anchor: [0.5, 1],
+          })
+        });
+      };
+
+      var clusterSource = function (records = $scope.geocoded_data) {
+        return new ol.source.Cluster({
+          source: buildingSources(records),
+          distance: 45,
+        });
+      };
+
+      var pointsLayerStyle = function(feature) {
+        var size = feature.get('features').length;
+        if (size > 1) {
+          return clusterPointStyle(size)
+        } else {
+          return singlePointStyle();
+        }
+      };
+
+      $scope.points_layer = new ol.layer.Vector({
+        source: clusterSource(),
+        zIndex: 2,  // Note: This is used for layer toggling.
+        style: pointsLayerStyle
+      });
+
+      // Hexbin layer
+      var hexagon_size = 750;
```

Comment:
I'd like to revisit this size after seeing other/different datasets.

Reply:
Given, UAT will be done with different datasets, this would be a good point to be mindful of during that.